scripts/create_dummy_profiles_json.py

```diff
@@ -0,0 +1,358 @@
+"""
+This script creates two JSON data files:
+
+1. JSON data file that is suitable for importing to the Helsinki profile service.
+   For more details, see open-city-profile project, models.Profile.import_customer_data
+
+2. JSON data file that contains the customer profiles
+   See CustomerProfileManager.import_customer_data for details.
+
+NOTE: the second file needs additional processing before importing.
+CustomerProfile needs to have UUID of the profile in the Helsinki profile service,
+and the UUID is generated by the Helsinki profile service at import time.
+When file (1) is imported, the Helsinki profile admin UI view returns a JSON file containing
+a mapping customer_id -> actual UUID
+
+api_url - use to fetch data about berths, piers and harbors. Running this script only reads data from the API,
+it does not change data in the server. The data is used to:
+* assign dummy customers to berths that are actually available
+* ensure dummy boats that fit in their berth
+
+-e flag can be used to specify the email address(es) to use for dummy users.
+If we actually send lots of emails while testing, then the
+recipient address should probably be a real email address, not something random.
+Mailgun (email service used as of 2020-12) has spammer detection logic and if it sees a lot of
+undeliverable mail, it might decide we are up to no good. "Mailgun has thresholds in place that if
+exceeded will result in a domain being temporarily disabled."
+
+Last name of the generated profiles starts with "Berth" so that the generated profiles
+can be identified in the database.
+
+"""
+
+import argparse
+import json
+import os
+import random
+import string
+import sys
+import traceback
+import uuid
+
+from faker import Faker
+from gql import Client, gql
+from gql.transport.requests import RequestsHTTPTransport
+
+NAME_CHARS = string.ascii_letters + "äöåÄÖÅ "
+EMAIL_CHARS = string.ascii_letters + "."
+
+fake = Faker("fi_FI")
+
+
+def randstr(chars=NAME_CHARS, min_len=1, max_len=10):
+    return "".join(
+        random.choice(chars) for _ in range(random.choice(range(min_len, max_len + 1)))
+    )
+
+
+HELSINKI_PROFILE_TEMPLATE = {
+    "customer_id": lambda: str(uuid.uuid4()),
+    "first_name": fake.first_name,
+    "last_name": lambda: "Berth" + fake.last_name(),
+    "email": randstr(min_len=10, max_len=10)
+    + "@"
+    + randstr(min_len=10, max_len=10)
+    + ".com",
```

Comment:
https://faker.readthedocs.io/en/master/providers/faker.providers.internet.html#faker.providers.internet.Provider.company_email

Reply:
I discussed this yesterday with Kalle, he suggested we don't generate real-looking emails, because we might generate a real email by accident. If a real person gets a test invoice by accident, it looks like a phishing scam... So we ended up with a design that normally dummy users would be generated with the email(s) supplied on the command line and if fake ones are requested, then generate really random emails that have almost no chance of matching a real email.